core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -85,6 +87,8 @@ public class HTTPClient implements RESTClient {
 
   private HTTPClient(
       String uri,
+      HttpHost proxy,
+      CredentialsProvider credsProvider,
```

Comment:
I think this should be named `proxyCredentialsProvider` to clearly indicate that this is only configured when the proxy is configured